mathics/builtin/drawing/discrete_plot.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+"""
+Discrete Plot
+"""
+import numbers
+
+from functools import lru_cache
+from typing import Optional
+
+from mathics.builtin.drawing.plot import (
+    Plot,
+    check_plot_range,
+    compile_quiet_function,
+    get_plot_range,
+)
+from mathics.builtin.graphics import Graphics
+from mathics.builtin.options import options_to_rules
+
+from mathics.core.attributes import A_HOLD_ALL, A_PROTECTED
+from mathics.core.convert.python import from_python
+from mathics.core.expression import Expression
+from mathics.core.list import ListExpression
+from mathics.core.symbols import Symbol, SymbolList
+from mathics.eval.nevaluator import eval_N
+from mathics.eval.plot import eval_ListPlot
+
+
+class _DiscretePlot(Plot):
+
+    attributes = A_HOLD_ALL | A_PROTECTED
+
+    expect_list = False
+
+    messages = {
+        "prng": (
+            "Value of option PlotRange -> `1` is not All, Automatic or "
+            "an appropriate list of range specifications."
+        ),
+        "invexcl": (
+            "Value of Exclusions -> `1` is not None, Automatic or an "
+            "appropriate list of constraints."
+        ),
+    }
+
+    options = Graphics.options.copy()
+    options.update(
+        {
+            "Axes": "True",
+            "AspectRatio": "1 / GoldenRatio",
+            "PlotRange": "Automatic",
+            "$OptionSyntax": "Strict",
+        }
+    )
+
+    rules = {
+        # One argument-form of DiscretePlot
+        "DiscretePlot[expr_, {var_Symbol, nmax_Integer}]": "DiscretePlot[expr, {var, 1, nmax, 1}]",
+        # One argument-form of DiscretePlot
+        "DiscretePlot[expr_, {var_Symbol, nmin_Integer, nmax_Integer}]": "DiscretePlot[expr, {var, nmin, nmax, 1}]",
+    }
+
+    def eval(self, function, x, start, nmax, step, evaluation, options):
+        """DiscretePlot[function_, {x_Symbol, start_Integer, nmax_Integer, step_Integer},
+        OptionsPattern[DiscretePlot]]"""
+        if isinstance(function, Symbol) and function.name is not x.get_name():
+            rules = evaluation.definitions.get_ownvalues(function.name)
+            for rule in rules:
+                functions = rule.apply(function, evaluation, fully=True)
+
+        if function.get_head_name() == "List":
```

Comment:
Can't this be converted to a get_head() == SymbolList?

Reply:
Good catch. Even better is `function.head is SymbolList`. Fixed now (and in the code where this was copied from).

It is amazing how much rot we find in the code when a new section is looked at.